intro-to-git/commits.md

```diff
@@ -0,0 +1,290 @@
+
+When a new file is created in a project, it is _untracked._ No matter how many times we change the contents of an untracked file, Git will never be able to understand the diff between versions.
+
+The command to begin tracking (watching for changes) a new file is the same command to bring it to staging:
+
+```
+$ git add <relative-path-to-untracked-file>
+```
+
+Where `<relative-path-to-untracked-file>` is replaced with a relative path to the file.
+
+## Commits Are Made After Local and Staging
+
+Programmers create a commit _after their intended changes are in **staging.**_
+
+Every commit contains the following information:
+
+- The summary of lines of code that were added, deleted, or changed, compared to the last commit.
+- A way to refer to the last commit/its "parent" commit
+- A unique ID for the commit, often called the commit hash. This gets generated automatically
+- A commit message, which will be a string that describes the commit's contents. Programmers write this at the time of building a commit
+- The date and time that the commit was made
+
+We create a commit with the command `$ git commit`. However, it is universally common to write the commit message at the time of making commit.
+
+The command to make a commit and write a commit message is:
+
+```
+$ git commit -m "Some string that describes the changes, don't forget the quotes"
+```
+
+Where `"Some string that describes the changes, don't forget the quotes"` is replaced with an accurate message.
+
+## About Commit Messages
+
+For each commit, the commit message should describe the changes found in the commit.
+
+There are no rules about commit messages, but these are common qualities of good commit messages:
+
+- short; can be a sentence fragment, a sentence, or two
+- accurate
+- specific
+- starts with a verb
+- finishes the sentence, "When we apply this commit, this commit will..."
+
+Specificity in commit messages helps you and your teammates understand the Git history when reviewing it. Most importantly, this helps you in the future, when you need to debug something and find what commit contains the bug. Specific commit messages could be difference between five hours or five minutes of debugging.
+
+### Example Commit Messages
+
+Some example commit messages include:
+
+- `"Fix failing RouteFinder tests"`
+- `"Reworks hikes and mountain searching logic"`
+- `"Add tests for hike guide meta-data"`
+- `"Update pytest dependency and generate test config"`
+- `"Renames class HikeFinder to HikePlanner"`
+
+Many software teams will designate a template to follow for commit messages.
+
+### My Terminal Gets Funky With Commit Messages
+
+Working with commit messages can often bring us to other text editor programs. By default, many commands around git commits use the vim program. The program vim is really great at displaying text! However, vim uses unique navigation keys.
+
+When a git commit command opens up the vim program, we navigate by:
```

Comment:
I'd love to know if you'd push back on keeping it as "the vim program" -- Since "vim" is a new term, I'd love for it to annoyingly stick that vim is a program... in my opinion, the word "vim" is so unfamiliar with non-programming vernacular, I kind of want some intense "this is what vim is"

Reply:
From a flow of language standpoint, I'd really prefer it be left off. But it did certainly do its job of drawing attention to it being a program, to the point that it felt too conspicuous to me.  I was just thinking that if this were something like Chrome, or Word, would it still make sense to state it as 'the Chrome program' or 'the Word program'?  I'm tempted to say no, we would just say Chrome or Word.  Students will be encountering the names of many new programs as they progress, so it felt strange to single out vim in just this particular context.

If I had my druthers, the language would be as I edited it, and we would call out command line program names by using inline code formatting.  But that would be a massive change.  So for this, or any other suggestion that I make, I fully trust that you will consider the parts that reinforce the effect you intended, and reject or modify those that take it in a different direction!